.github/workflows/website.yml

```
@@ -48,30 +68,77 @@ jobs:
       - uses: bahmutov/npm-install@v1
         env:
           PLAYWRIGHT_SKIP_BROWSER_DOWNLOAD: 1
-      - run: npm run build -w packages/website
+      - run: touch .env
+      - run: |
+          (
+            npm -w '@web3-storage/website' run ci:envfile --silent \
+            | sed "s/^NEXT_PUBLIC_MAGIC=$/NEXT_PUBLIC_MAGIC=$NEXT_PUBLIC_MAGIC/" \
+            | sed "s/^MAGIC_SECRET_KEY=$/MAGIC_SECRET_KEY=$MAGIC_SECRET_KEY/" \
+            ;
+            echo 'NEXT_PUBLIC_MAGIC_TESTMODE_ENABLED="1"';
+          ) > .env
         env:
-          NEXT_PUBLIC_API: http://localhost:8787
-          NEXT_PUBLIC_ENV: test-e2e
           NEXT_PUBLIC_MAGIC: ${{ secrets.STAGING_MAGIC_PUBLIC_KEY }}
+          MAGIC_SECRET_KEY: ${{ secrets.STAGING_MAGIC_SECRET_KEY }}
+      - name: load db schema
+        run: |
+          npm -w '@web3-storage/db' run start
+          npm -w '@web3-storage/db' run load-schema
+          npm -w '@web3-storage/db' run stop
+      - run: node -r dotenv/config $(which npm) run build -w packages/website
       - run: npx playwright install --with-deps ${{ matrix.browser }} chromium
         working-directory: packages/website
-      - run: touch .env
-      - run: xvfb-run --auto-servernum --server-args="-screen 0 1280x960x24" -- npm run -w @web3-storage/website test:e2e -- --project=${{ matrix.browser }}
+      - id: website_test_e2e
+        run: xvfb-run --auto-servernum --server-args="-screen 0 1280x960x24" -- node -r dotenv/config $(which npm) run -w @web3-storage/website test:e2e -- --project=${{ matrix.browser }} --reporter=html
         env:
-          NEXT_PUBLIC_API: http://localhost:8787
-          NEXT_PUBLIC_ENV: dev
-          NEXT_PUBLIC_MAGIC: ${{ secrets.STAGING_MAGIC_PUBLIC_KEY }}
-      - name: Check for website test results
+          PLAYWRIGHT_TIMEOUT: 120000
+
+      - if: always()
+        name: Check for website playwright-report
+        id: check_website_playwright_report
+        uses: andstor/file-existence-action@v1
+        with:
+          files: packages/website/playwright-report
+
+      - name: add playwright-report to web3.storage
+        if: inputs.add_playwright_report_to_web3_storage && steps.check_website_playwright_report.outputs.files_exists == 'true'
+        uses: web3-storage/add-to-web3@v2
+        id: ipfs
+        with:
+          path_to_add: packages/website/playwright-report
+          web3_token: ${{ secrets.WEB3_TOKEN }}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: log playwright-report on web3.storage cid, url
+        if: inputs.add_playwright_report_to_web3_storage && steps.check_website_playwright_report.outputs.files_exists == 'true'
+        run: |
+          echo cid: ${{ steps.ipfs.outputs.cid }}
+          echo url: https://nftstorage.link/ipfs/${{ steps.ipfs.outputs.cid }}
```

even newer! even shinier!
```suggestion
          echo url: https://w3s.link/ipfs/${{ steps.ipfs.outputs.cid }}
```

---

done